include/envoy/server/instance.h

```diff
@@ -220,7 +220,7 @@ class Instance {
   virtual std::chrono::milliseconds statsFlushInterval() const PURE;
 
   /**
-   * Flush the server stats
+   * Flush the stats sinks outside of a flushing interval.
```

Comment:
sounds good; was just wondering whether this restarted the interval or just was an extra flush that got injected. Probably doesn't matter that much. E.g. if the interval is 5s, and I call flushStats() at 8s, would I get flushes at 5,8,10,15, or at 5,8,13 ?

Also, does this block until the stats are flushed? Or does it queue it up to happen next?

Reply:
Yep, I see what you were asking. This only changes the access modifiers on the already existing `flushStats()`[1](https://github.com/envoyproxy/envoy/blob/9105aead0f5c89419bd2aca789b3738967c0baa1/source/server/server.cc#L173) method in the InstanceImpl. To answer your questions about the implementation:
- The call re-enables the `stats_flush_timer_` so 5,8,13.
-  Before initialization this call blocks because histograms are skipped. However, after envoy is fully initialized the flushing gets passed as the `PostMergeCb` for `mergeHistograms`.

Reply:
@jmarantz let me know if you have any other concerns

Reply:
sg; I don't know how important it is to doc that the interval timer is reset. However, I think it would be good to note that the interval still continues in some form.

And I also think doccing that this only requests an flush to happen 'soon' and the admin request will get a response before the flush completes.

Also the new admin-endpoint needs actual doc in the rst file.

Reply:
sorry, I think I am confused. I am not exposing this to an admin endpoint?

Reply:
My fault for trying to review PRs from vacation without deeply looking :)

Can you add a unit test? Is there going to be a call-site in OSS in a future PR?

Reply:
Oh no! I am so sorry, I didn't remember you were out. My bad, I'll stop pestering. I thought `TEST_P(ServerStatsTest, FlushStats)` already covered it. But thanks to you I re-read and realized that the test was advancing time! Well, now we can just flushStats() :)

re: call-site. Not immediately needed in Envoy itself, but useful for Envoy Mobile (https://github.com/lyft/envoy-mobile/issues/573)

Reply:
Also clarify here that flushStats() will return prior to the stats actually getting flushed, in case that matters to a caller. If it does, we might need to add a version that calls a lambda when the stat-flush is complete.

Reply:
Good point! I am going to think about this a bit more.

Reply:
Thought some more and synced with the Envoy Mobile team. We think this is ok for now but leaves room for future work. https://github.com/lyft/envoy-mobile/issues/700

Thanks for all the discussion here, really helpful!